lib/sycamore/sycamore/connectors/opensearch/opensearch_reader.py

```diff
@@ -45,7 +45,7 @@ def read_records(self, query_params: BaseDBReader.QueryParams) -> "OpenSearchRea
         assert "index" not in query_params.kwargs and "body" not in query_params.kwargs
         if "scroll" not in query_params.kwargs:
             query_params.kwargs["scroll"] = "1m"
-        if "size" not in query_params.kwargs:
+        if "size" not in query_params.query:
```

Comment:
Scratching my head here. Why are you checking it in `.query` but setting it in `.kwargs` ?

Reply:
The first is trying to see if the user provided the size param in the query itself, the latter is trying to pass in a default value without modifying the query. I don't love it but I'm scared to touch it right now